Implemented OnBoarding Wizard

JWKaiqi opened the pull request:
Issue ga4gh/dockstore#743 and ga4gh/dockstore#810 has been updated.

- Updated the setup page, which now includes the on-boarding wizard.
- Also fixed the text overflow problem for the starred page, when the title in the starred tools and workflows page is too long, the panel truncate the title with "..."

garyluu commented on src/app/app.module.ts:
```diff
@@ -70,6 +71,8 @@ import { StarringModule } from './starring/starring.module';
 import { ListentryModule } from './listentry/listentry.module';
 import {AdvancedsearchComponent} from './search/advancedsearch/advancedsearch.component';
+import { DownloadcliclientComponent } from './loginComponents/onboarding/downloadcliclient/downloadcliclient.component';
+import { SetupcompleteComponent } from './loginComponents/onboarding/setupcomplete/setupcomplete.component';
```
Try to use upper camel case for component names.  (i.e. SetupCompleteComponent)

garyluu commented on src/app/loginComponents/token.service.ts:
```diff
@@ -26,4 +26,49 @@ export class TokenService {
   }
 
+  getWebServiceVersion() {
+    const url = `${ Dockstore.API_URI }/api/ga4gh/v1/metadata`;
```
this already exists in versions.service.ts, merge the two

garyluu commented on src/app/loginComponents/token.service.ts:
```diff
@@ -26,4 +26,49 @@ export class TokenService {
+  }
+
+  getUserTokens(userId) {
+    const url = `${ Dockstore.API_URI }/users/${userId}/tokens`;
```
this is also available in user.service.ts

garyluu commented on src/app/loginComponents/onboarding/onboarding.component.html:
```diff
@@ -1,4 +1,26 @@
+  <nav class="pull-right">
+    <ul class="pager">
+      <li *ngIf="curStep > 1">
+        <a style="cursor: pointer" (click)="prevStep()"><span class="glyphicon glyphicon-chevron-left"></span>Previous</a>
```
avoid inline styles when possible

garyluu replied:
On second thought...perhaps a button would make more sense.  That would include the pointer by default.  @denis-yuen , any reasons why this component in UI1 is not a button?

JWKaiqi replied:
@garyluu need to modify the css if we use a button instead.

denis-yuen replied:
@garyluu  No idea, try it out

denis-yuen commented on src/app/app.module.ts:
```diff
@@ -89,7 +91,9 @@ import {AdvancedsearchComponent} from './search/advancedsearch/advancedsearch.co
     AuthComponent,
     TokensComponent,
     StarredentriesComponent,
-    AdvancedsearchComponent
```
Hmmm,  `StarredEntriesComponent` and `AdvancedSearchComponent` for consistency?

JWKaiqi commented:
Updated pull request:
- merged duplicated functions into one service.
- changed <a> to a <button> for the navigation buttons of the OnBording Wizard page (i.e: previous, next, and finish)
- changed component names to upper case (StarredEntriesComponent and AdvancedSearchComponent).
